tests/testsuite/install.rs

```diff
@@ -2730,3 +2730,20 @@ fn uninstall_running_binary() {
 
 "#]]).run();
 }
+
+#[cargo_test]
+fn dry_run() {
+    pkg("foo", "0.0.1");
+
+    cargo_process("-Z unstable-options install --dry-run foo")
+        .masquerade_as_nightly_cargo(&["install::dry-run"])
+        .with_stderr_data(str![[r#"
+[UPDATING] `dummy-registry` index
+[DOWNLOADING] crates ...
+[DOWNLOADED] foo v0.0.1 (registry `dummy-registry`)
+[INSTALLING] foo v0.0.1
+
+"#]])
+        .run();
+    assert_has_not_installed_exe(paths::cargo_home(), "foo");
```

Comment:
`cargo test` should pass on every commit.  So instead, this should either be a test without `--dry-run` and then ext commit adds it, or this errors about the flag not existing.

Reply:
I've rebased my fisrt commit so it test the output without the `--dry-run` flag, the second commit change this test and add the `--dry-run` flag to the options so we can see the diff in git

Reply:
Thanks!